Data Structures/HashMap.java

```diff
@@ -0,0 +1,153 @@
+import java.util.Scanner;
+public class HashMap {
+ static class Pair {
+        int key;
+        int value;
+        Pair next;
+        Pair(int key, int value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+    }
+    static class hashMap {
+        int size;
+        Pair[] arr;
+
+        hashMap(int size) {
+            this.size = size;
+            arr = new Pair[size];
+            for (int i = 0; i < size; i++) {
+                arr[i] = null;
+            }
+        }
+
+        //  Function to insert a key value pair in the hash map
+        public int hashFunction(int key) {
+            return key % size;
+        }
+
+        //  Function to insert a key value pair in the hash map
+        public void insert(int key, int value) {
+            int index = hashFunction(key);
+            Pair newNode = new Pair(key, value);
+            if (arr[index] == null) {
+                arr[index] = newNode;
+            } else {
+                Pair temp = arr[index];
+                while (temp.next != null) {
+                    temp = temp.next;
+                }
+                temp.next = newNode;
+            }
+        }
+
+        //  Function to search a key in the hash map
+        public void delete(int key) {
+            int index = hashFunction(key);
+            Pair temp = arr[index];
+            if (temp.key == key) {
+                arr[index] = temp.next;
+            } else {
+                while (temp.next != null) {
+                    if (temp.next.key == key) {
+                        temp.next = temp.next.next;
+                        break;
+                    }
+                    temp = temp.next;
+                }
+            }
+        }
+
+        //  Function to search a key in the hash map
+        public int get(int key) {
+            int index = hashFunction(key);
+            Pair temp = arr[index];
+            while (temp != null) {
+                if (temp.key == key) {
+                    return temp.value;
+                }
+                temp = temp.next;
+            }
+            return -1;
+        }
+
+        public void display() {
```

Comment:
purpose-full-variable name ,you have to conscious about it.

Reply:
okay sir